src/api/v2/ResourceGroupsExtensionManager.ts

```diff
@@ -1,9 +1,22 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
 import * as vscode from 'vscode';
 
 interface ResourceGroupsContribution {
     readonly activation?: {
         readonly onFetch?: string[];
     }
+    readonly application: {
+        readonly branches?: { type: string }[];
+        readonly resources?: { type: string }[];
+    }
+    readonly workspace: {
+        readonly branches?: { type: string }[];
+        readonly resources?: { type: string }[];
+    }
```

Comment:
For application branches and resources, I think type should be an `AzExtResourceType`.

For workspace branches and resources, it can be an arbitrary string.

Reply:
I don't disagree necessarily, but I think I'd like to address this as a follow up PR so as to not get mixed up with all the rest, and give me time to think about this.